charging.cpp

```diff
@@ -0,0 +1,11 @@
+#include "charging.hpp"
+
+std::string getCurrentRangeOccurance(int* ptr)
```

Comment:
This function takes two integers and makes a range out of them. Make it take two integers and return a `struct` representing a range.
Next, write a test to concatenate ranges. Combining them will solve the problem.
This is one way to solve the problem. There are other ways to. Go ahead and try to word the problem using tests.